crates/core/transaction/src/view.rs

```diff
@@ -94,6 +101,174 @@ impl TransactionView {
     pub fn action_views(&self) -> impl Iterator<Item = &ActionView> {
         self.body_view.action_views.iter()
     }
+
+    /// Acts as a higher-order translator that summarizes a TransactionSummary by consolidating
+    /// effects for each unique address.
+    pub fn accumulate_effects(summary: &TransactionSummary) -> TransactionSummary {
```

Comment:
I think we should just have this be private, and call it internally

Reply:
good catch — this was intended to be private